rust/src/conf.rs

```diff
@@ -92,23 +84,20 @@ pub fn conf_get_bool(key: &str) -> bool {
 /// Wrap a Suricata ConfNode and expose some of its methods with a
 /// Rust friendly interface.
 pub struct ConfNode {
-    pub conf: *const c_void,
+    pub conf: *const SCConfNode,
 }
 
 impl ConfNode {
```

Comment:
@catenacyber This is one of those interesting API cases..

As the `Conf` API is in C, it makes sense for the raw bindings to be in the sys crate, and a nicer API exposed in the `suricata` crate, or even the `suricata_core` crate.

This gets weird when we have something like `JsonBuilder`, which is pure Rust, exported to C, with that C re-exported back to Rust.  Where does the nice wrapper belong for it?  We'd always have it in `suricata`, or `suricata_core`.  Do we need another name-spaces for such Rust items that have gone from Rust to C back to Rust for the same of plugins library user? `suricata_extern` or something like that?